src/Erl/Data/Map.purs

```diff
@@ -56,18 +59,36 @@ fold = foldMImpl ((#))
 -- | Fold the keys and values of a map, accumulating values using some
 -- | `Monoid`.
 foldMap :: forall a b m. Monoid m => (a -> b -> m) -> Map a b -> m
-foldMap f = fold (\acc k v -> acc <> f k v) mempty
+foldMap f = fold (\acc k v -> f k v <> acc) mempty
 
 -- | Fold the keys and values of a map, accumulating values and effects in
 -- | some `Monad`.
 foldM :: forall a b m z. Monad m => (z -> a -> b -> m z) -> z -> Map a b -> m z
 foldM f z = foldMImpl bind f (pure z)
 
+-- | Convert any foldable collection of key/value pairs to a map.
+-- | On key collision, later values take precedence over earlier ones.
+fromFoldable :: forall f k v. Ord k => Foldable f => f (Tuple k v) -> Map k v
+fromFoldable = foldl (\m (Tuple k v) -> insert k v m) empty
+
+--foldr :: forall a b. (a -> b -> b) -> b -> f a -> b
+--foldl :: forall a b. (b -> a -> b) -> b -> f a -> b
+--foldMap :: forall a m. Monoid m => (a -> m) -> f a -> m
 instance foldableMap :: Foldable (Map a) where
-  foldl f = fold (\z _ -> f z)
   foldr f z m = foldr f z (values m)
+  foldl f = fold (\z _ -> f z)
   foldMap f = foldMap (const f)
 
+
+--foldrWithIndex :: forall a b. (i -> a -> b -> b) -> b -> f a -> b
```

Comment:
what's with the commented types? can we delete or use actual type annotations - you can annotate the instance implementations if need be